### PR TITLE
[SERVER-1940] remove gcp vm service networking rules

### PR DESCRIPTION
**Pull request by lloydChris**
# Description

Cleans up whitespace

One can use custom subnetting in GCP ,this removes the recommendation that it is not possible

We have been recommending customers put k8s nodes, Nomad Clients and VMs for machine jobs in separate subnets.  
We we also suggesting three firewall rules to accompany the separate network.  The `source` for the firewall rules was overlapping and no guidance was provided on how to deal with that.  This removes those recommendations

There is still the recommendation that customers keep everything behind a private VPC and use a VPN to connect, which is a good first step in securing the networking.

The information needed to build out the firewall rules between subnets is still provided in the "hardening your cluster" section, which is now linked from here.

The bottom line is customer environments are all unique in terms of structure and security requirements, it is difficult to even reference a similar starting point in GCP because of the project inheritance and hierarchy.  Lets remove anything that is ambiguous and empower the customer to make the decisions that are right for them (by linking to the Hardening your Cluster docs)

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

**Review by atulsingh0 — APPROVED** [review]

File: jekyll/_cci2/server/installation/phase-3-execution-environments.adoc [hunk 1]
```diff
@@ -1000,27 +1000,6 @@ databaseEncryption:
...
-----
-+
-[source,shell]
-----
-gcloud compute firewall-rules create "circleci-vm-service-internal-k8s-fw" --network "<network>" --action allow --source-ranges "<clusterIpv4Cidr>" --rules "TCP:22,TCP:2376"
-----
-+
-[source,shell]
-----
-gcloud compute firewall-rules create "circleci-vm-service-external-fw" --network "<network>" --action allow --rules "TCP:54782"
```
Comment (circlery) [inline jekyll/_cci2/server/installation/phase-3-execution-environments.adoc:1021]: I agree with the removal of the above two, but won't the removal of 54782 result in customers not being able to SSH-rerun into VM-based jobs?

**Review by devinmarieb — APPROVED** [review]

File: jekyll/_cci2/server/installation/phase-3-execution-environments.adoc [hunk 2]
```diff
@@ -1004,24 +1004,8 @@ databaseEncryption:
...
-----
+
+External VMs need the networking rules described in [Hardening your Cluster](https://circleci.com/docs/server/installation/hardening-your-cluster/#external-vms)
```
Comment (lloydChris) [inline jekyll/_cci2/server/installation/phase-3-execution-environments.adoc:1008]: @rosieyohannan i will defer to you on the best was to build this link, i suspect there is some internal path referencing that would be prefered

**Comment by rosieyohannan** [comment]
@lloydChris I've updated that link. I think this PR is good to merge now? Only issue is that I will need to copy the changes over to the v4.1 docs in the 4.1 release branch because we're now going to be duplicating the docs for each major and minor release, and I've already done the duplication for 4.1 .... I wonder if the easiest thing would be to incorporate these changes into the 4.1 release branch? Or would you rather they be published ASAP?

**Comment by lloydChris** [comment]
Yes it is good to merge.

There is no rush to get this published ASAP.  so whatever is most convenient for your workflow.  I suspect it would be easiest to merge to main, then cherry pick onto version specific branches?

**Review by circlery — APPROVED** [review]

**Pull request merged**